packages/block/src/from-rpc.ts

```diff
@@ -28,8 +30,7 @@ function normalizeTxParams(_txParams: any) {
   // v as raw signature value {0,1}
   // v is the recovery bit and can be either {0,1} or {27,28}.
   // https://ethereum.stackexchange.com/questions/40679/why-the-value-of-v-is-always-either-27-11011-or-28-11100
```

Comment:
We could update this comment in light of: https://ethereum.stackexchange.com/questions/113499/why-is-value-of-ecdsa-v-often-neither-27-nor-28

Reply:
So I'm still stumped about this. I've been looking at `tx` trying to understand what `v` should be when it's passed in as part of the `fromTxData` constructor we need to use when pulling transactions from the provider.  If I understand things, v should always be { 0, 1, 27, 28 }.  For legacy transactions, we have this `meets155` helper that checks the to make sure it matches the legacy expectations and for Type 1 and 2 transactions, it should always be 0 or 1. So, to my way of thinking, our "from-rpc" helper that formats the data should just take the value of v that's passed in and let the `tx` constructors error out if something's wrong, right?  I shouldn't need to do any special massaging of data passed in from the RPC endpoint since those should just be formatting helpers to make sure that values that come in as hex strings are converted to bigints or what have you, right?

Reply:
So there are 3 Ethereum "periods" for the v values:

1. Before [EIP-155](https://eips.ethereum.org/EIPS/eip-155): `v` was `27` or `28` back then (up to `SpuriousDragon` HF, block `2,675,000`).
2. After `EIP-155`: v has now the chain ID baked in according to the `EIP-155` formula `{0,1} + CHAIN_ID * 2 + 35`. So the minimal `v` is `0 + 1*2 +35` which is `37`
3. Along with typed txs introduced with [EIP-2718](https://eips.ethereum.org/EIPS/eip-2718) (first concrete one: `EIP-2930` with these access lists introduced in the `Berlin` HF (block `12,244,000`) `v` went back to simply `0` and `1` (so what the signature ECDSA function expects as an input) and is should technically also not called `v` any more but something like `yParity` or something (but still is for RPC for backwards-compatibility reasons)

So: if we keep this code with `v + 27` we are killing off the typed tx v values. If I am not totally overlooking something we should update here, so basically - as Andrew suggested - just letting the `v` values "through".

I think this `v + 27` code was for some very old Geth bug behavior where it returned (I think in the very old per-SpuriousDragon days) `0` or `1` instead of `27` or `28` in RPC (so - attention! - big source of confusion if one stumbles upon such a transaction: values which are *now*  correct but which weren't back in the days).

Since this bug has been fixed a long time ago I think we can safely upgrade here (again: just passing the `v` values through).

Reply:
Great! Thanks for the detailed explanation holger!